Assets/MixedRealityToolkit.SDK/Inspectors/Utilities/Solvers/SurfaceMagnetismInspector.cs

```diff
@@ -0,0 +1,112 @@
+        {
+            base.OnEnable();
+
+            magneticSurfacesProperty = serializedObject.FindProperty("magneticSurfaces");
+            maxDistanceProperty = serializedObject.FindProperty("maxRaycastDistance");
+            closestDistanceProperty = serializedObject.FindProperty("closestDistance");
+            surfaceNormalOffsetProperty = serializedObject.FindProperty("surfaceNormalOffset");
+            surfaceRayOffsetProperty = serializedObject.FindProperty("surfaceRayOffset");
+            currentRaycastDirectionModeProperty = serializedObject.FindProperty("currentRaycastDirectionMode");
+            raycastModeProperty = serializedObject.FindProperty("raycastMode");
+            boxRaysPerEdgeProperty = serializedObject.FindProperty("boxRaysPerEdge");
+            orthographicBoxCastProperty = serializedObject.FindProperty("orthographicBoxCast");
+            maximumNormalVarianceProperty = serializedObject.FindProperty("maximumNormalVariance");
+            sphereSizeProperty = serializedObject.FindProperty("sphereSize");
+            volumeCastSizeOverrideProperty = serializedObject.FindProperty("volumeCastSizeOverride");
+            useLinkedAltScaleOverrideProperty = serializedObject.FindProperty("useLinkedAltScaleOverride");
+            currentRaycastDirectionModeProperty = serializedObject.FindProperty("currentRaycastDirectionMode");
+            orientationModeProperty = serializedObject.FindProperty("orientationMode");
+            orientationBlendProperty = serializedObject.FindProperty("orientationBlend");
+            debugEnabledProperty = serializedObject.FindProperty("debugEnabled");
+
+            surfaceMagnetism = target as SurfaceMagnetism;
+        }
+
+        public override void OnInspectorGUI()
+        {
+            base.OnInspectorGUI();
+
+            serializedObject.Update();
+
+            // General Properties
+            EditorGUILayout.LabelField("General Properties", EditorStyles.boldLabel);
+            EditorGUILayout.PropertyField(surfaceNormalOffsetProperty);
+            EditorGUILayout.PropertyField(surfaceRayOffsetProperty);
+            EditorGUILayout.PropertyField(orientationModeProperty);
+            EditorGUILayout.PropertyField(orientationBlendProperty);
+
+            // Raycast properties
+            EditorGUILayout.LabelField("Raycast Properties", EditorStyles.boldLabel);
+            EditorGUILayout.PropertyField(magneticSurfacesProperty, true);
+            EditorGUILayout.PropertyField(closestDistanceProperty);
+            EditorGUILayout.PropertyField(maxDistanceProperty);
+            EditorGUILayout.PropertyField(currentRaycastDirectionModeProperty);
+            EditorGUILayout.PropertyField(raycastModeProperty);
+
+            if (raycastModeProperty.enumValueIndex != (int)SceneQueryType.SphereOverlap)
+            {
+                if (raycastModeProperty.enumValueIndex != (int)SceneQueryType.SimpleRaycast)
+                {
```

Comment:
Consider deleting this line, since you check for the specific things you care about down below

OR

Alternatively, this is actually a pretty good candidate for a switch statement like:

switch (raycastModeProperty.enumValueIndex):
case (int)SceneQueryType.SphereOverlap:
   InspectorUIUtility.DrawWarning("SurfaceMagnetism does not support SphereOverlap raycast mode");
   break;
case (int)SceneQueryType.BoxRaycast:
   EditorGUILayout.PropertyField(boxRaysPerEdgeProperty);
   EditorGUILayout.PropertyField(orthographicBoxCastProperty);
   EditorGUILayout.PropertyField(maximumNormalVarianceProperty);
   break;
...

It puts the code assocaited with the condtiion really close by (especially for the SphereOverlap case, where the warning message is currently pretty far from the if check)

Reply:
This is a nit btw, it's just usually when I see a  lot of if statements it can be kinda like owaw

---
In reply to: [308997604](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/5416#discussion_r308997604) [](ancestors = 308997604)